src/core/connection.c

```diff
@@ -6567,6 +6660,217 @@ QuicConnParamSet(
         return QUIC_STATUS_SUCCESS;
     }
 
+    case QUIC_PARAM_CONN_ADD_LOCAL_ADDRESS: {
+
+        if (BufferLength != sizeof(QUIC_ADDR)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (QuicConnIsServer(Connection)) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        if (!Connection->State.Started) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
```

Comment:
Why not? It seems kind of intuitive to me that a client could create a connection, queue all paths it wants to use and then start it. No?

Reply:
I agree with you. So, I've also implemented this. https://github.com/masa-koz/msquic/pull/4

Reply:
Now, we can add local address before connection start. And I've added tests for adding/removing address.
https://github.com/microsoft/msquic/pull/4218/commits/2b3c6fd450f8a93746e3bfd8a03d00d7c34aeafc